process/publisherHandler.go

```diff
@@ -0,0 +1,193 @@
+package process
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-ws-connector-firehose-go/data"
+	"github.com/multiversx/mx-chain-ws-connector-firehose-go/data/hyperOutportBlocks"
+)
+
+type publisherHandler struct {
+	handler               HyperBlockPublisher
+	outportBlocksPool     BlocksPool
+	dataAggregator        DataAggregator
+	retryDuration         time.Duration
+	firstCommitableBlocks map[uint32]uint64
+
+	blocksChan chan []byte
+	cancelFunc func()
+	closeChan  chan struct{}
+}
+
+// NewPublisherHandler creates a new publisher handler component
+func NewPublisherHandler(
+	handler HyperBlockPublisher,
+	outportBlocksPool BlocksPool,
+	dataAggregator DataAggregator,
+	retryDurationInMiliseconds uint64,
+	firstCommitableBlocks map[uint32]uint64,
+) (*publisherHandler, error) {
+	if check.IfNil(handler) {
+		return nil, ErrNilPublisher
+	}
+	if check.IfNil(outportBlocksPool) {
+		return nil, ErrNilBlocksPool
+	}
+	if check.IfNil(dataAggregator) {
+		return nil, ErrNilDataAggregator
+	}
+
+	ph := &publisherHandler{
+		handler:               handler,
+		outportBlocksPool:     outportBlocksPool,
+		dataAggregator:        dataAggregator,
+		retryDuration:         time.Duration(retryDurationInMiliseconds) * time.Millisecond,
+		firstCommitableBlocks: firstCommitableBlocks,
+		blocksChan:            make(chan []byte),
+		closeChan:             make(chan struct{}),
+	}
+
+	var ctx context.Context
+	ctx, ph.cancelFunc = context.WithCancel(context.Background())
+
+	go ph.startListener(ctx)
+
+	return ph, nil
+}
+
+func (ph *publisherHandler) startListener(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			log.Debug("closing commonPublisher listener")
+			return
+		case headerHash := <-ph.blocksChan:
+			ph.handlePublishEvent(headerHash)
+		}
+	}
+}
+
+// PublishHyperBlock will push aggregated outport block data to the firehose writer
+func (ph *publisherHandler) PublishBlock(headerHash []byte) error {
+	select {
+	case ph.blocksChan <- headerHash:
+	case <-ph.closeChan:
+	}
```

Comment:
is this even needed? why not simply `ph.blocksChan <- headerHash`? then remove closeChan and the for loop inside handlePublishEvent

Reply:
added closeChan so then external calls will not hang if this component will be closed